test/test.c

```diff
@@ -21324,6 +21324,22 @@ void test_SSLHandshakeFirst(void)
     s = natsOptions_TLSHandshakeFirst(opts);
     testCond(s == NATS_OK);
 
+    test("Set TLSHandshakeFirst option without setting secure: ");
+    {
+        // we start with a new natsOptions struct so that we can test
+        // that it does not crash with a minimal config
+        natsOptions *opts = NULL;
+        s = natsOptions_Create(&opts);
+        IFOK(s, natsOptions_SetURL(opts, "nats://127.0.0.1:4443"));
+        IFOK(s, natsOptions_SetTimeout(opts, 500));
+        s = natsOptions_TLSHandshakeFirst(opts);
+        IFOK(s, natsConnection_Connect(&nc, opts));
+        testCond(s == NATS_SSL_ERROR);
```

Comment:
So to be clear, we are expecting an error because some other SSL needed options were not set (like skip verify hostname).

Reply:
yes adding a comment.